spatial/r2/deprecated.go

```diff
@@ -0,0 +1,40 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r2
```

Comment:
```
// TODO: Delete this file for v0.10.0.
```

and in the other deprecated.go as well (you can put me in the TODO parens if you want).